go-controller/hybrid-overlay/pkg/controller/master.go

```diff
@@ -0,0 +1,263 @@
+package controller
+
+import (
+	"bytes"
+	"fmt"
+	"net"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/hybrid-overlay/pkg/types"
+	houtil "github.com/ovn-org/ovn-kubernetes/go-controller/hybrid-overlay/pkg/util"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kube"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/allocator"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+
+	kapi "k8s.io/api/core/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+)
+
+// MasterController is the master hybrid overlay controller
+type MasterController struct {
+	kube      *kube.Kube
+	allocator *allocator.SubnetAllocator
+}
+
+// NewMaster a new master controller that listens for node events
+func NewMaster(clientset kubernetes.Interface) (*MasterController, error) {
+	m := &MasterController{
+		kube:      &kube.Kube{KClient: clientset},
+		allocator: allocator.NewSubnetAllocator(),
+	}
+
+	// Add our hybrid overlay CIDRs to the allocator
+	for _, clusterEntry := range config.HybridOverlay.ClusterSubnets {
+		err := m.allocator.AddNetworkRange(clusterEntry.CIDR.String(), 32-clusterEntry.HostSubnetLength)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Mark existing hostsubnets as already allocated
+	existingNodes, err := m.kube.GetNodes()
+	if err != nil {
+		return nil, fmt.Errorf("Error in initializing/fetching subnets: %v", err)
+	}
+	for _, node := range existingNodes.Items {
+		if houtil.IsWindowsNode(&node) {
```

Comment:
@danwinship yeah. I think the non-Windows stuff is orthogonal, because those will actually be full Kube nodes and not really hybrid-overlay nodes. There are really two parts: (1) the non-Linux node subnet assignment and node annotation munging on the master, and (2) the actual vxlan tunnel on the nodes themselves. Windows stuff does (1) and (2), while the non-Windows stuff will be mostly (2).